lib/DxilContainer/DxilContainerAssembler.cpp

```diff
@@ -1527,7 +1527,8 @@ void hlsl::SerializeDxilContainerForModule(DxilModule *pModule,
   std::unique_ptr<DxilPSVWriter> pPSVWriter = nullptr;
   unsigned int major, minor;
   pModule->GetDxilVersion(major, minor);
-  RootSignatureWriter rootSigWriter(pModule->GetSerializedRootSignature());
+  RootSignatureWriter rootSigWriter(std::move(pModule->GetSerializedRootSignature())); // Grab RS here
```

Comment:
Per standard, the only valid thing to do on a moved-out object is to destroy it. Even your empty() below is undefined, as well as anything else we might try to do with that vector. It'd have been better to `std::swap` it with your empty one.

Also I'd have liked a comment explaining why we are leaving the module without a root signature.